Add template query (#9119)
opensearch-project/documentation-website: _query-dsl/specialized/template.md

---
layout: default
title: Template
parent: Specialized queries
nav_order: 70
---

# Template query
Introduced 2.19
{: .label .label-purple }

> Use a `template` query to create search queries that contain placeholder variables. Placeholders are specified using the `"${variable_name}"` syntax (Note that the variables need to be double-quoted). When you submit a search request, these placeholders remain unresolved until they are processed by search request processors. This approach is particularly valuable when your initial search request contains data that needs to be transformed or generated at runtime.

Reviewer: ```suggestion
Use a `template` query to create search queries that contain placeholder variables. Placeholders are specified using the `"${variable_name}"` syntax (note that the variables must be enclosed in quotation marks). When you submit a search request, these placeholders remain unresolved until they are processed by search request processors. This approach is particularly valuable when your initial search request contains data that needs to be transformed or generated at runtime.
```
